tools/onnx-subgraph/Readme.md

```diff
@@ -0,0 +1,86 @@
+# onnx_autosubgraph
+onnx-subgraph tool provides  model auto partitionioning of onnx model to several sub models by operator, performance and model size limitations, 
+with the order and input / output names of sub models
+
+# How to build the onnx-subgraph
+## OS environment dependence
+     1. ubuntu >=20.04
+     2. GCC >= 9.4.0
+     3. cmake >= 3.10
+     4. python >= 3.8
+     5. apt-get install libprotobuf-dev protobuf-compiler
+
+## Python packages dependence
+    onnx                         1.16.0
+    onnxruntime                  1.18.1
+    onnxsim                      0.4.36
+    torch                        2.3.1
+    scikit-image
+    scikit-learn
+    pandas
+    tqdm
+    
+## building the onnx-subgraph
+    1. cd onnx-subgraph
+    2. bash 3rd_files_download.sh
+    3. mkdir build & cd build
+    4. cmake .. & make
+    5. we can get following output at ./build
+          ├── onnx-subgraph
+          └── scripts
+              ├── config.json
+              ├── config-sample-1.json
+              ├── config-sample-2.json
+              ├── extract_onnx_lib.py
+              ├── extract_onnx.py
+              ├── model_inference_multiple_output.py
+              ├── model_inference.py
+              ├── onnx_subgraph_ut.py
+              ├── quant.py
+              ├── single_vs_multiple_onnx.py
+              └── test_model_download.sh
+# How to use the onnx-subgraph
+## Pre-steps
+### Download the test AI models
+    1. bash scripts/test_model_download.sh, then "resnet-test.onnx" will be got in ./build
+    2. you can change to any other onnx files as your needs, or edit the download link in "scripts/test_model_download.sh"
```

Comment:
if this line exceeds 100 cols, please split so that will be within 100 cols.
plz apply to all other codes